pkg/common/peertracker/uds_bsd.go

```diff
@@ -12,9 +15,71 @@ func getCallerInfo(fd uintptr) (CallerInfo, error) {
 		return CallerInfo{}, err
 	}
 
+	p, err := process.NewProcess(int32(result))
+	if err != nil {
+		return CallerInfo{}, err
+	}
+
+	uID, err := getUID(p)
+	if err != nil {
+		return CallerInfo{}, err
+	}
+
+	gID, err := getGID(p)
+	if err != nil {
+		return CallerInfo{}, err
+	}
+
+	// Addr expected to fail on k8s when "hostPID" is not provided
+	addr, _ := getAddr(p)
+
 	info := CallerInfo{
-		PID: int32(result),
+		PID:        int32(result),
+		UID:        uint32(uID),
+		GID:        uint32(gID),
+		BinaryAddr: addr,
 	}
 
 	return info, nil
 }
+
+func getUID(proc *process.Process) (int32, error) {
+	uids, err := proc.Uids()
+	if err != nil {
+		return 0, fmt.Errorf("failed UIDs lookup: %v", err)
+	}
+
+	switch len(uids) {
+	case 0:
+		return 0, fmt.Errorf("failed UIDs lookup: no UIDs for process")
+	case 1:
+		return uids[0], nil
+	default:
+		return uids[1], nil
+	}
+}
+
+func getGID(proc *process.Process) (int32, error) {
+	gids, err := proc.Gids()
+	if err != nil {
+		return 0, fmt.Errorf("failed GIDs lookup: %v", err)
+	}
+
+	switch len(gids) {
+	case 0:
+		return 0, fmt.Errorf("failed GIDs lookup: no GIDs for process")
+	case 1:
+		return gids[0], nil
+	default:
+		return gids[1], nil
+	}
+}
+
+func getAddr(proc *process.Process) (string, error) {
+	path, err := proc.Exe()
```

Comment:
This seems to be unimplemented in gopsutil: https://github.com/shirou/gopsutil/blob/76779af9093add061eca53ec76ae4116acbf8a4c/process/process_freebsd.go#L67

Which seems strange, since this information should be available as part of `ki_args`: https://github.com/freebsd/freebsd-src/blob/d0ea5e467f3c44909667c5ee90c3d26653fb6687/sys/sys/user.h#L124